tests/integration/it_utils.py

```diff
@@ -86,8 +126,31 @@ def _choose_client_async(use_json_client: bool) -> Client:
         return ASYNC_WEBSOCKET_CLIENT
 
 
+def _get_client(is_async: bool, is_json: bool, is_testnet: bool) -> Client:
+    if is_testnet:
+        if is_async:
+            if is_json:
+                return ASYNC_JSON_RPC_TESTNET_CLIENT
+            else:
+                return ASYNC_WEBSOCKET_TESTNET_CLIENT
+        else:
+            if is_json:
+                return JSON_RPC_TESTNET_CLIENT
+            else:
+                return WEBSOCKET_TESTNET_CLIENT
+    else:
+        if is_async:
+            return _choose_client_async(is_json)
+        else:
+            return _choose_client(is_json)
```

Comment:
removing these extraneous `else`s will make this much easier to read

```
def _get_client(....):
  if is_testnet:
    if is_async:
      if is_json:
        return ASYNC_JSON_RPC_TESTNET_CLIENT
      return ASYNC_WEBSOCKET_TESTNET_CLIENT
   if is_json:
      return JSON_RPC_TESTNET_CLIENT
   return WEBSOCKET_TESTNET_CLIENT
   
   if is_async:
     return _choose_client_async(is_json)
   return _choose_client(is_json)
```

but even better would be helper methods

Reply:
you could do this with a dictionary whose keys are bitmasks

```
# [is_testnet, is_async, is_json]
_CLIENT_CHOICES = {
  0x111: ASYNC_JSON_RPC_TESTNET_CLIENT,
  0x110: ASYNC_WEBSOCKET_TESTNET_CLIENT,
  ....
}

def _get_client(is_async: bool, is_json: bool, is_testnet: bool) -> Client:
  bit_array = [int(value) * 10 ** (i+1) for i, value in enumerate([is_json, is_async, is_testnet])]
  bit_combination = functools.reduce(operator.or_, bit_array)
  return _CLIENT_CHOICES[bit_combination]
  
```

I'm sure there's a more elegant way to do the bitwise stuff in python than what i've done, but you get what i mean